src/main/game/Board.java

```diff
@@ -3,78 +3,67 @@
 import java.util.ArrayList;
 import java.util.List;
 
-import static java.util.Arrays.asList;
-import static main.game.Marks.O;
-import static main.game.Marks.X;
+import static main.game.Mark.*;
 import static main.game.Result.TIE;
 
 public class Board {
 
-    public final int ROW_COUNT = 3;
-    public List<String> grid;
+    private int size;
+    public List<Mark> grid;
 
-    public Board(List<String> grid) {
+    public Board(int size, List<Mark> grid) {
+        this.size = size;
         this.grid = grid;
     }
 
-    public Board() {
-        this.grid = asList("1", "2", "3", "4", "5", "6", "7", "8", "9");
+    public Board(int size) {
+        this.size = size;
+        this.grid = createGrid();
     }
 
-    public List updateMove(String move, String playerType) {
-        int convertedMove = Integer.parseInt(String.valueOf(move)) - 1;
-        this.grid.set(convertedMove, playerType);
-        return this.grid;
-    }
-
-    public List<List<String>> rows() {
-        List<List<String>> rows = new ArrayList<>();
-
-        for (int i = 0; i < ROW_COUNT * ROW_COUNT; i += 3) {
-            List<String> row = asList(grid.get(i), grid.get(i + 1), grid.get(i + 2));
-            rows.add(row);
+    private List<Mark> createGrid() {
+        List<Mark> spaces = new ArrayList<>();
+        for (int i = 0; i < size * size; i++) {
+            spaces.add(EMPTY);
         }
+        return spaces;
+    }
 
-        return rows;
+    public List<Mark> updateMove(int position, Mark mark) {
+        this.grid.set(position - 1, mark);
+        return this.grid;
     }
 
     public List<Line> rowLines() {
         List<Line> rows = new ArrayList<>();
-
-        for (int i = 0; i < ROW_COUNT * ROW_COUNT; i += 3) {
-            Line line = new Line(grid.get(i), grid.get(i + 1), grid.get(i + 2));
-            rows.add(line);
+        for (int i = 0; i < size * size; i += size) {
+            if (size == 3) {
```

Comment:
Checking the size gets you through today, but what about 5x5? 6x6? It adds a lot of code.
There has to be some mathy trick! For example:
If you number each row and column individually e.g. for 3x3, you'd get rows 0, 1, 2 and columns 0, 1, 2.
One of your diagonals is ALWAYS on same pairs: (0,0), (1,1), (2,2)...

Maybe that helps?